client/src/Dashboard/Dashboard.jsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import '../index.css';
+import { authenticationRepository, accountRepository } from '@/_services';
+import config from 'config';
+const signalR = require('@aspnet/signalr')
+
+class Dashboard extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            accountId: null,
+            currentPlan: null,
+            numberOfUsers: null,
+            maximumNumberOfUsersAllowed: null,
+            pricePerMonth: null,
+            hub: null
+        };
+
+        this.logout = this.logout.bind(this);
+        this.upgradePlan = this.upgradePlan.bind(this);
+    }
+
+    componentDidMount() {
+        const hub = this.configureUserHub();
+        this.start(hub);
+
+        this.getAccount();
+    }
+
+    getAccount() {
+        accountRepository.get(this.props.location.state.id)
+            .then(createdAccount => {
+                this.setState({ accountId: createdAccount.id,
+                    currentPlan: createdAccount.currentPlan, 
+                    numberOfUsers: createdAccount.users.length, 
+                    maximumNumberOfUsersAllowed: createdAccount.maximumNumberOfUsersAllowed,
+                    pricePerMonth: createdAccount.pricePerMonth })
+        });
+    }
+
+    start(hub) {
+        this.setState({ hub: hub }, () => {
+            this.state.hub.start()
+
+            this.state.hub.on("AddedUser", () => {
+                this.setState({ numberOfUsers: this.state.numberOfUsers + 1 })
+            });
+        });
+    }
+
+    configureUserHub() {
+        return new signalR.HubConnectionBuilder()
+            .withUrl(`${config.apiUrl}/userHub`)
+            .build();
+    }
+
+    renderUserLimitExceededMessage() {
+        if (this.state.numberOfUsers == this.state.maximumNumberOfUsersAllowed) {
+            return (
+                <div className="alert is-error">You have exceeded the maximum number of users for your account, please upgrade your plan to increase the limit.</div>
+            );
+        }
+    }
+
+    renderUpgradeButton() {
+        if (this.state.currentPlan == 1) {
+            return (
+                <button onClick={this.upgradePlan} className="button is-success">Upgrade to Enterprise Plan</button>
+            );
+        }
+    }
+
+    upgradePlan() {
+        accountRepository.upgradePlan(this.state.accountId);
+        this.setState({ currentPlan: 2, maximumNumberOfUsersAllowed: 1000, pricePerMonth: 1000});
+    }
+
+    renderHeader() {
+        if (this.state.currentPlan == 1) {
```

Comment:
Small nit: Identifying a plan just by the integer can be confusing and lead to bugs. Personally, I would prefer this to be a string which would carry more meaning with it.

As a side effect, it would also eliminate this entire method:
```jsx
<header>${this.state.currentPlan} Plan - ${this.state.pricePerMonth}/Month</header>
```

Reply:
Created an enum for them so it makes more sense!